tests/usermetas.test.js

```diff
@@ -1,21 +1,108 @@
 'use strict';
 
+const {
+	assert
+} = require('chai');
 const Helper = require('../server/users/helpers');
+const router = require('../server/users/index');
 
-let userId = 'test1';
 
 describe('usermetas', async () => {
-    describe('#dynamicMetadata()', () => {
-      it('returns user metas based on a test user', async () => {
-        const helper = td.object(Helper);  
-  
-        td.when(helper.dynamicMetadata(userId, 'location,locale,auth', td.callback))
-          .thenCallback(null, 'test');
-  
-        
-      });
-   
-   
-    });
-  });
-  
\ No newline at end of file
+	describe('#dynamicMetadata()', () => {
+		it('returns user metas based on a test user', (done) => {
+			const userId = 'test1';
+			const metadataList = ['usermeta1', 'usermeta2'];
+			const helper = td.object(Helper);
+			const response = [{
+				id: "usermeta1",
+				value: ""
+			}, {
+				id: "usermeta2",
+				value: ""
+			}];
+
+			td.when(helper.dynamicMetadata(userId, metadataList.join(','), td.callback))
+				.thenCallback(null, response);
+
+			const res = td.object(['json']);
+			const req = {
+				params: {
+					username: userId
+				}
+			};
+			const next = function(err) {
+				try {
+					assert(err == undefined, 'there should not be an error');
+					td.verify(res.json(response));
+					done();
+				} catch (e) {
+					done(e);
+				}
+			};
+			router.getUsermetaMiddleware(helper, metadataList.join(','))(req, res, next);
+		});
+
+		it('checking metadatalist emtpy', (done) => {
+			const userId = 'test1';
+			const metadataList = [];
+			const helper = td.object(Helper);
+			const response = [{
+				id: "",
+				value: ""
+			}];
+
+			td.when(helper.dynamicMetadata(userId, metadataList.join(','), td.callback))
+				.thenCallback(null, response);
+
+			const res = td.object(['json']);
+			const req = {
+				params: {
+					username: userId
+				}
+			};
+			const next = function(err) {
+				try {
+					assert(err == undefined, 'there should not be an error');
+					td.verify(res.json(response));
+					done();
+				} catch (e) {
+					done(e);
+				}
+			};
+			router.getUsermetaMiddleware(helper, metadataList.join(','))(req, res, next);
+		});
+
+    it('Updating metadata', (done) => {
+			const userId = 'test1'; 
+      const metaKey = 'usermeta1';
+      const metaVal = 'from-tests';
+			const helper = td.object(Helper);
+			const response = {};
+
+			td.when(helper.updateDynamicUserMeta(userId, metaKey, metaVal, td.callback))
+				.thenCallback(null, response);
+
+			const res = td.object(['json']);
+			const req = {
+				params: {
+					username: userId, 
+				},
+        body: {
```

Comment:
Careful with indentation.

Don't bother fixing those for now - there is a way to do it automatically, I'll run it on the whole project once all Pull Requests (PR) are merged into develop.

Reply:
I am not sure why the indentation looks very bad here ... 
while check the below photo its from the VS.. it looks very fine .. what do you think ?

<img width="1134" alt="Screen Shot 2021-11-07 at 9 47 54 PM" src="https://user-images.githubusercontent.com/11071646/140657813-7ed857d5-c667-4246-8122-57bb0dbfd6f9.png">